unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/VoiceChatHUD/VoiceChatWindowController.cs

```diff
@@ -0,0 +1,436 @@
+        voiceChatBarView.OnLeaveVoiceChat -= LeaveVoiceChat;
+        dataStore.player.otherPlayers.OnAdded -= OnOtherPlayersStatusAdded;
+        dataStore.player.otherPlayers.OnRemoved -= OnOtherPlayerStatusRemoved;
+        ownProfile.OnUpdate -= OnUserProfileUpdated;
+        friendsController.OnUpdateFriendship -= OnUpdateFriendship;
+        settings.generalSettings.OnChanged -= OnSettingsChanged;
+
+        currentPlayers.Clear();
+        playersPool.Clear();
+    }
+
+    internal void CloseView() { SetVisibility(false); }
+
+    internal void JoinVoiceChat(bool isJoined)
+    { 
+        using (var iterator = currentPlayers.GetEnumerator())
+        {
+            while (iterator.MoveNext())
+            {
+                iterator.Current.Value.SetAsJoined(isJoined);
+            }
+        }
+
+        voiceChatWindowView.SetAsJoined(isJoined);
+
+        if (isJoined)
+        {
+            voiceChatBarView.Show();
+            SetWhichPlayerIsTalking();
+        }
+        else
+        {
+            dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
+            isOwnPLayerTalking = false;
+            voiceChatBarView.Hide();
+        }
+
+        this.isJoined = isJoined;
+
+        if (!isJoined)
+        {
+            MuteAll(true);
+            socialAnalytics.SendVoiceChannelDisconnection();
+        }
+        else
+        {
+            MuteAll(voiceChatWindowView.isMuteAllOn);
+            socialAnalytics.SendVoiceChannelConnection(currentPlayers.Count);
+        }
+
+        dataStore.voiceChat.isJoinedToVoiceChat.Set(isJoined);
+    }
+
+    internal void LeaveVoiceChat() { JoinVoiceChat(false); }
+
+    internal void GoToCrowd() { WebInterface.GoToCrowd(); }
+
+    internal void OnOtherPlayersStatusAdded(string userId, Player player)
+    {
+        if (!currentPlayers.ContainsKey(player.id))
+        {
+            var otherProfile = UserProfileController.userProfilesCatalog.Get(player.id);
+
+            if (otherProfile != null)
+            {
+                VoiceChatPlayerComponentView elementView = null;
+                if (playersPool.Count > 0)
+                    elementView = playersPool.Dequeue();
+                else
+                {
+                    elementView = voiceChatWindowView.CreateNewPlayerInstance();
+                    elementView.OnMuteUser += MuteUser;
+                    elementView.OnContextMenuOpen += OpenContextMenu;
+                }
+
+                elementView.Configure(new VoiceChatPlayerComponentModel
+                {
+                    userId = otherProfile.userId,
+                    userImageUrl = otherProfile.face256SnapshotURL,
+                    userName = otherProfile.userName,
+                    isMuted = false,
+                    isTalking = false,
+                    isBlocked = false,
+                    isFriend = friendsController.GetFriends().ContainsKey(userId),
```

Comment:
i think you should check: `friendsController.ContainsStatus(userId, FriendshipStatus.FRIEND)`. You could have a request open and it would not be valid

Reply:
Ih thanks for the warning! I thought the `GetFriends()` method only contained the already accepted friends. Changed!

Done in https://github.com/decentraland/unity-renderer/pull/2346/commits/4cb873c4445c9c6127fc3b80ed37c0bce9b50a13.